### ISSUE TITLE
[Core] Ray init memory issue with docker

### ISSUE THREAD

**Issue by Milad137** [ray-project/ray#24149]
### What happened + What you expected to happen

When running ray in a docker container, if the memory size is not defined in docker run, "/sys/fs/cgroup/memory.max" file sets it as max instead of a integer number. This leads to ray.init() to fail, as it tries to get memory size.
https://github.com/ray-project/ray/blob/e402fc0eaa07d3dd396730661bb799e5aab9a03a/python/ray/_private/utils.py#L394

### Versions / Dependencies

Python 3.9.7
Ubuntu 20.04
Ray 1.12.0

### Reproduction script

import ray

ray.init()

`
--------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/tmp/ipykernel_7829/2135109301.py in <module>
      1 import ray
      2 
----> 3 ray.init()
      4 
      5 

/opt/conda/lib/python3.9/site-packages/ray/_private/client_mode_hook.py in wrapper(*args, **kwargs)
    103             if func.__name__ != "init" or is_client_mode_enabled_by_default:
    104                 return getattr(ray, func.__name__)(*args, **kwargs)
--> 105         return func(*args, **kwargs)
    106 
    107     return wrapper

/opt/conda/lib/python3.9/site-packages/ray/worker.py in init(address, num_cpus, num_gpus, resources, object_store_memory, local_mode, ignore_reinit_error, include_dashboard, dashboard_host, dashboard_port, job_config, configure_logging, logging_level, logging_format, log_to_driver, namespace, runtime_env, storage, _enable_object_reconstruction, _redis_max_memory, _plasma_directory, _node_ip_address, _driver_object_store_memory, _memory, _redis_password, _temp_dir, _metrics_export_port, _system_config, _tracing_startup_hook, **kwargs)
   1020         # handler. We still spawn a reaper process in case the atexit handler
   1021         # isn't called.
-> 1022         _global_node = ray.node.Node(
   1023             head=True, shutdown_at_exit=False, spawn_reaper=True, ray_params=ray_params
   1024         )

/opt/conda/lib/python3.9/site-packages/ray/node.py in __init__(self, ray_params, head, shutdown_at_exit, spawn_reaper, connect_only)
    299 
    300         if not connect_only:
--> 301             self.start_ray_processes()
    302             # we should update the address info after the node has been started
    303             try:

/opt/conda/lib/python3.9/site-packages/ray/node.py in start_ray_processes(self)
   1128         # Make sure we don't call `determine_plasma_store_config` multiple
   1129         # times to avoid printing multiple warnings.
-> 1130         resource_spec = self.get_resource_spec()
   1131         (
   1132             plasma_directory,

/opt/conda/lib/python3.9/site-packages/ray/node.py in get_resource_spec(self)
    470                 resources,
    471             ) = merge_resources(env_resources, self._ray_params.resources)
--> 472             self._resource_spec = ResourceSpec(
    473                 self._ray_params.num_cpus if num_cpus is None else num_cpus,
    474                 self._ray_params.num_gpus if num_gpus is None else num_gpus,

/opt/conda/lib/python3.9/site-packages/ray/_private/resource_spec.py in resolve(self, is_head, node_ip_address)
    195 
    196         # Choose a default object store size.
--> 197         system_memory = ray._private.utils.get_system_memory()
    198         avail_memory = ray._private.utils.estimate_available_memory()
    199         object_store_memory = self.object_store_memory

/opt/conda/lib/python3.9/site-packages/ray/_private/utils.py in get_system_memory()
    434     elif os.path.exists(memory_limit_filename_v2):
    435         with open(memory_limit_filename_v2, "r") as f:
--> 436             docker_limit = int(f.read())
    437 
    438     # Use psutil if it is available.

ValueError: invalid literal for int() with base 10: 'max\n'
`

### Issue Severity

Low: It annoys or frustrates me.

**Comment by xwjiang2010** [comment]
Can you also add which version of Ray you are using?

**Comment by richardliaw** [comment]
This should be fixed with #23922, which should come out in ray 1.13! 

Otherwise you can try hacking around this:

```
import ray
import psutil
psutil_memory_in_bytes = psutil.virtual_memory().total

ray._private.utils.get_system_memory = lambda: psutil_memory_in_bytes
```

**Comment by Milad137** [comment]
Thanks for your help. That was actually what I was doing. it's not a big issue though.

**Comment by Milad137** [comment]
> This should be fixed with #23922, which should come out in ray 1.13!
> 
> Otherwise you can try hacking around this:
> 
> ```
> import ray
> import psutil
> psutil_memory_in_bytes = psutil.virtual_memory().total
> 
> ray._private.utils.get_system_memory = lambda: psutil_memory_in_bytes
> ```

Thanks for your help. That was actually what I was doing. it's not a big issue though.

**Comment by Milad137** [comment]
This should be fixed with https://github.com/ray-project/ray/pull/23922, which should come out in ray 1.13!
